main.js

```diff
@@ -0,0 +1,236 @@
+    function success(position) {
+      let latitude  = position.coords.latitude;
+      let longitude = position.coords.longitude;
+      fetchCurrent(latitude, longitude);
+      fetchForecast(latitude, longitude);
+    }
+
+    function error() {
+      alert('Unable to retrieve your location');
+    }
+
+    navigator.geolocation.getCurrentPosition(success, error);
+  };
+
+  // API call to get current weather data from OpenWeatherMap based on the user's query.
+  let fetchCurrent = function (query) {
+
+    let formattedQueryCurrent = 'http://api.openweathermap.org/data/2.5/weather?q=' + query + ',us&units=imperial&appid=0b3fcf70b1228659daf9b5a171c0e96f';
+    if (arguments.length === 2) {
+      let latitude = arguments[0];
+      let longitude = arguments[1];
+      formattedQueryCurrent = 'http://api.openweathermap.org/data/2.5/weather?lat=' + latitude + '&lon=' + longitude + '&units=imperial&appid=0b3fcf70b1228659daf9b5a171c0e96f';
+    }
+    // Check that the format of the user's query is correct
+    console.log(formattedQueryCurrent);
+    // Request current weather data
+    $.ajax({
+      method: "GET",
+      url: formattedQueryCurrent,
+      dataType: "json",
+
+      success: function(data) {
+        addCurrentData(data);
+      },
+
+      error: function(jqXHR, textStatus, errorThrown) {
+        console.log(textStatus);
+      }
+    });
+  };
+
+  // API call to get weather forecast data from OpenWeatherMap based on the user's query.
+  let fetchForecast = function (query) {
+    let formattedQueryForecast = 'http://api.openweathermap.org/data/2.5/forecast?q=' + query + ',us&units=imperial&appid=0b3fcf70b1228659daf9b5a171c0e96f';
+    if (arguments.length === 2) {
+      let latitude = arguments[0];
+      let longitude = arguments[1];
+      formattedQueryForecast = 'http://api.openweathermap.org/data/2.5/forecast?lat=' + latitude + '&lon=' + longitude + '&units=imperial&appid=0b3fcf70b1228659daf9b5a171c0e96f';
+    }
+    // Request 5-day forecast data
+    $.ajax({
+      method: "GET",
+      url: formattedQueryForecast,
+      dataType: "json",
+
+      success: function(data) {
+        addForecastData(data);
+      },
+      complete: function() {
+        $('.set-default').removeClass('invisible');
+      },
+      error: function(jqXHR, textStatus, errorThrown) {
+        console.log(textStatus);
+      }
+    });
+  };
+
+  return {
+    saveToLocalStorage,
+    getFromLocalStorage,
+
+    clearWeather,
+    renderWeather,
+
+    addCurrentData,
+    addForecastData,
+
+    useCurrentLocation,
+
+    fetchCurrent,
+    fetchForecast
+  }
+}
+
+let weatherApp = new CurrentWeather();
+
+// Event handlers
+$('.search').on( 'click', function () {
+  weatherApp.clearWeather();
+  console.log('clicked');
+  var search = $('#search-query').val();
+  weatherApp.fetchCurrent(search);
+  weatherApp.fetchForecast(search);
+});
+
+$('.search-current-location').on( 'click', function () {
+  weatherApp.useCurrentLocation();
```

Comment:
A better design here would be to abstract out the getting of the location here and then only pass the data to the app if the current location was successful.